### ISSUE TITLE
Transaction resulting with TRANSACTION_EXPIRED after a given time

### ISSUE THREAD

**Issue by failfmi** [hashgraph/hedera-sdk-go#313]
### Description

On `mainnet`, now occasionally, when I submit a transaction to Hedera, it resolves after around 5-10 minutes, with `TRANSACTION_EXPIRED`. Of course, this is happening, because the transaction's valid start becomes in the past, thus making it invalid. 

I've tried to dig in how the submission of transactions is done in the sdk and what troubles me is that when you pick a node, and it's **not** healthy, you do not switch to the next one, but you delay your current selected for some time and stay on it.

Is there any way this to be avoided?

### Steps to reproduce

1. Submitting a `SubmitTopicConsensusMessage` transaction
2. Submitting a `ScheduleCreate/Sign` transaction

### Hedera network

mainnet

### Version

v2.1.5

### Operating system

Linux

**Comment by failfmi** [comment]
Given that the transaction official resolves with an error after 5-10mins, does this mean that it has passed through different nodes for 5mins, could not submit it to any one of them, and finally has found one and submitted it? Or did it stay on the first one the whole time.

The way the transaction is executed is the following
```go
messageTxId, err := ts.hederaNode.SubmitTopicConsensusMessage(topicID, msgBytes)
```

**Comment by failfmi** [comment]
Given that the transaction **officially** resolves with an error after 5-10mins, does this mean that it has passed through different nodes for 5mins, could not submit it to any one of them, and finally has found one and submitted it? Or did it stay on the first one the whole time.

The way the transaction is executed is the following:
```go
txResponse, err := hedera.NewTopicMessageSubmitTransaction().
		SetTopicID(topicId).
		SetMessage(message).
		Execute(client)
```

**Comment by SimiHunjan** [comment]
@danielakhterov @andrix10 please investigate

**Comment by danielakhterov** [comment]
@failfmi I think the reason this is happening is not because of how the SDK does exponential backoff or how it uses nodes. I believe the reason this is happening is because nodes 4 and 10 on mainnet are misbehaving right now and don't resolve to a connection for 5m. I thought Go already had this fixed (cc @andrix10 ), but from the looks of it, it doesn't. We have recently done similar fixes in both Java and JS.
Note: I'd argue this issue isn't with the SDKs, but rather with gRPC itself; it does not give us a way to set a timeout for connecting and hence we need to hack it in ourselves.

> Is there any way this to be avoided?

For now you can remove nodes 4 and 10 from the network.

**Comment by steven-sheehy** [comment]
Can't you pass in a Go Context into the grpc layer that will timeout in a generic fashion?

**Comment by jsindy** [comment]
Also, what time was this happening on nodes 4 and 10? Is it still happening? Can you describe how we can reproduce this on our end? Thanks.

**Comment by danielakhterov** [comment]
> Can't you pass in a Go Context into the grpc layer that will timeout in a generic fashion?

Not sure, I haven't looked into it myself, but I'm just working off what I know about the other SDKs which required had to hack around this issue.

> Also, what time was this happening on nodes 4 and 10? Is it still happening? Can you describe how we can reproduce this on our end? Thanks.

I believe it is still the case (which is sort of good because it allows us an easy way to test our fixes.)

**Comment by Daniel-K-Ivanov** [comment]
Here are the records with timestamps of when the errors occurred:
```
ERRO[2021-11-10T09:26:41.201650586Z] [0.0.9262-1636488993-336667306] - Failed to submit schedule sign [0.0.570061]. Error: [exceptional precheck status TRANSACTION_EXPIRED received for transaction 0.0.538148@1636535655.759549756].  context="Scheduled Service"

ERRO[2021-11-10T09:27:50.151566273Z] [0.0.563856-1636503359-943699923] - Failed to submit schedule sign [0.0.570062]. Error: [exceptional precheck status TRANSACTION_EXPIRED received for transaction 0.0.538148@1636535652.296263936].  context="Scheduled Service

ERRO[2021-11-10T09:27:50.662062113Z] [0.0.542193-1636496927-697597783] - Failed to submit schedule sign [0.0.570053]. Error: [exceptional precheck status TRANSACTION_EXPIRED received for transaction 0.0.538148@1636535650.232297334].  context="Scheduled Service

ERRO[2021-11-10T09:27:51.111497644Z] [0.0.542193-1636501846-015511291] - Failed to submit schedule sign [0.0.570054]. Error: [exceptional precheck status TRANSACTION_EXPIRED received for transaction 0.0.538148@1636535653.904495026].  context="Scheduled Service

ERRO[2021-11-10T09:30:34.999467088Z] [0.0.563856-1636505245-382208271] - Failed to submit scheduled transfer transaction. Error [exceptional precheck status TRANSACTION_EXPIRED received for transaction 0.0.538148@1636535643.787055233].  context="Scheduled Service"

ERRO[2021-11-10T11:09:53.785980204Z] [0.0.499056-1636542066-371870955] - Failed to submit Signature Message to Topic. Error: [exceptional precheck status TRANSACTION_EXPIRED received for transaction 0.0.538148@1636542083.301339383]  context="Transfers Service"

ERRO[2021-11-10T11:44:25.565611208Z] [0.0.499056-1636541738-106829598] - Failed to submit scheduled transfer transaction. Error [exceptional precheck status TRANSACTION_EXPIRED received for transaction 0.0.538148@1636541753.174105335].  context="Scheduled Service"
```

All of those transactions were submitted through the SDK a couple of minutes earlier:
For example this is the time where 5 was submitted:
```
INFO[2021-11-10T09:14:15.389007765Z] New Transaction with ID: [0.0.563856-1636505245-382208271]  context="[0.0.540219] Transfer Watcher"
```
You can notice that the time is `09:14:15` and the error for expiry is at `09:30:34` -> `16 minutes` the TX was trying to be submitted and after 16 minutes it was submitted but the `startTimestamp` of the signed payload was too far in the past (2 minutes have passed)

**Comment by SimiHunjan** [comment]
This will be resolved v2.6.0

**Comment by danielakhterov** [comment]
I believe this is resolved, but #329 is a similar issues with a larger goal.
